### ISSUE TITLE
Properties with Symbol keys are stripped from objects, even with passthrough()

### ISSUE THREAD

**Issue by dmeehan1968** [colinhacks/zod#2734]
TL;DR Properties with symbol keys are stripped from objects regardless of the object shape.  

`ZodRawShape` is defined as `[k: string]: ZodTypeAny` but according to Typescript rules, this [does not mean that a passed object should only have string keys](https://github.com/microsoft/TypeScript/issues/53265).  I think this is why Zod allows the object shape but then doesn't handle the presence of Symbol keys.

Therefore there should probably, at least, be a runtime exception to indicate that they are not handled.

```ts
const bar = Symbol('bar')

const Schema = z.object({
  foo: z.string(),
  [bar]: z.string()
})

const input: z.infer<typeof Schema> = {
  foo: 'foo',
  [bar]: 'bar',
}

const output = Schema.parse(input)
```

Results in:

```js
{
  foo: 'foo'
}
```

It seems that symbol keys are ignored in the object specification, as `z.string().default('bar')` also does not result in there being a property for `Symbol(bar)` with value `bar` in the output.

This also happens if the symbol property is included or omitted, and the schema has `.passthrough()` added.

**Comment by syyyr** [comment]
Hello, I can confirm, that this issue is still relevant - even with Zod v4, and `loose()`, Symbols are not preserved. The only way to preserve symbols is to use `z.custom()`, but that has its issues (one being it's not a native parser, so you lose the advantages of using `z.object()`)

**Comment by syyyr** [comment]
Our usecase for this is that we have this custom format called "CPON", which is essentially a superset of JSON. In this format, there are objects (we call those "Map"), same as in JSON, but then there are "IMap" values, which are objects with numerical keys, to save some bytes by getting rid of the quotes. When parsing these two into JS values, we have to tag them with a `Symbol` to preserve the type information (as JS only has string keys). We then want to use _Zod_ to confirm a specific shape of the object. However, there are two problems:
- We can't simply use `z.object()`, because it doesn't support checking the symbols.
- We can't use `z.object().and(z.custom(...))` to check the symbol by ourselves, because by the time parsing gets to the `and`, the object has its symbols stripped by `z.object()`.

The only solution that I found is to use `z.custom()` for the whole thing, but that has a few disadvantages:
- We can't use things like `.strict()`, on the resulting parser, because it is not of type `ZodObject` (unless we implement that for own parser).
- We also can't use our custom parser in unions (although you can work around that by using `and()` on the union).
- We have to check that the symbol is present by ourselves (which is not a big deal I guess).
- Error messages (issues) are a bit cumbersome to work with in the custom parser, although in v4 (with `.check()`), things have gotten a lot better.

Another thing I've found out is that `z.record()` actually does work with `Symbol` keys, as that one uses some introspection library. If `z.object()` used the same thing for iteration, then that would work out well, I think. But I guess the introspection library might not be as fast.

**Comment by syyyr** [comment]
I tested with 4.1.12:
```typescript
const some_parser = z.object({x: z.string()}).loose();
const MySymbol = Symbol('');
const my_object = {
	[MySymbol]: 'some_string',
	x: 'some_string'
};
const parsed_object = some_parser.parse(my_object);
console.log('parsed_object', '=', parsed_object);
console.log('parsed_object', '=', parsed_object[MySymbol]);
```
And symbols are still stripped from objects.

**Comment by syyyr** [comment]
I have just checked with 4.3.5 and the problem still persists:
```typescript
const some_parser = z.object({x: z.string()}).loose();
const MySymbol = Symbol('');
const my_object = {
	[MySymbol]: 'some_string',
	x: 'some_string'
};
const parsed_object = some_parser.parse(my_object);
console.log('parsed_object', '=', parsed_object);
console.log('parsed_object', '=', parsed_object[MySymbol]);
```

**Comment by syyyr** [comment]
Still reproducible on Zod v4.4.3 with the same script from https://github.com/colinhacks/zod/issues/2734#issuecomment-3767512036
